WikiPreview.Fluent.Plugin/WikiPreviewSearchApp.cs

```diff
@@ -108,59 +109,88 @@ public ValueTask LoadSearchApplicationAsync()
-                            PreviewImage = bitmapImageResult,
-                            DisplayedName = displayedName,
-                            ResultName = resultName,
-                            SearchedText = searchedText,
-                            Score = score,
-                            PageId = pageId,
-                            AdditionalInformation = additionalInfo
-                        };
-                        await channel.Writer.WriteAsync(wikiPreviewSearchResult).ConfigureAwait(false);
-                    }, cancellationToken).ContinueWith(_ => channel.Writer.Complete());
+                        }, cancellationToken)
+                        .ContinueWith(_ => channel.Writer.Complete(), CancellationToken.None);
                 else
                     channel.Writer.Complete();
-            });
+            }, CancellationToken.None);
 
             await foreach (WikiPreviewSearchResult item in channel.Reader.ReadAllAsync(cancellationToken))
                 yield return item;
         }
 
-        public ValueTask<ISearchResult> GetSearchResultForId(string serializedSearchObjectId)
+        public ValueTask<WikiPreviewSearchResult> GetSearchResultForId(string serializedSearchObjectId)
+        {
+            try
+            {
+                if (string.IsNullOrWhiteSpace(serializedSearchObjectId))
+                    return default;
+
+                string url = "https://en.wikipedia.org/w/api.php?action=query&prop=extracts|pageimages&pageids=" +
+                             serializedSearchObjectId +
+                             "&explaintext&exintro&pilicense=any&pithumbsize=100&format=json";
+
+                using var httpClient = new HttpClient();
+                Task<Wiki> task = httpClient.GetFromJsonAsync<Wiki>(url, _serializerOptions);
+                task.Wait();
+                PageView pageView = task.Result.Query.Pages.Values.First();
+                ValueTask<WikiPreviewSearchResult> wikiTask = GenerateSearchResult(pageView, pageView.Title);
+
+                return wikiTask;
+            }
+            catch (Exception)
+            {
+                return default;
+            }
+        }
+
+        private static async ValueTask<WikiPreviewSearchResult> GenerateSearchResult(PageView value,
+            string searchedText)
         {
-            return new();
+            string resultName = value.Extract;
+            string displayedName = value.Title;
+            double score = displayedName.SearchDistanceScore(searchedText);
+            string pageId = value.PageId.ToString();
+            string wikiUrl = WikiRootUrl + displayedName;
+            BitmapImageResult bitmapImageResult;
+
+            string additionalInfo = "";
+            if (!string.IsNullOrWhiteSpace(resultName))
+            {
+                using var reader = new StringReader(resultName);
+                additionalInfo = await reader.ReadLineAsync().ConfigureAwait(false) ?? resultName;
```

Comment:
Just for your info, there is nothing asynchronous in `StringReader` unless it reads from a file/memory stream.
If you press `F12` on `ReadLineAsync` method you will see that it simply do `Task.FromResult(ReadLine())` which means, that in this case it allocates a `Task` for no reason.
I suggest you changing to `ReadLine` method.

Reply:
Yeah, I had this doubt. Thanks for auto-clarification :)